subsys/nrf_security/src/drivers/Kconfig

```
@@ -4,56 +4,230 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
-config MBEDTLS_PSA_ACCEL_KEY_TYPE_SUPPORT
+# Legacy config that needs to be removed later. The Oberon PSA driver is now the fallback option
+# for all the crypto operations when hardware acceleration is not available.
+config PSA_CRYPTO_DRIVER_OBERON
+	# The Oberon driver is required to provide HKDF needed for protected storage and bultin keys
+	prompt "Oberon PSA driver" if !(TFM_PARTITION_PROTECTED_STORAGE || TFM_CRYPTO_BUILTIN_KEYS)
 	bool
 	default y
-	depends on MBEDTLS_PSA_ACCEL_KEY_TYPE_ECC_KEY_PAIR || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_ECC_PUBLIC_KEY || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_RSA_KEY_PAIR || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_RSA_PUBLIC_KEY || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_AES || \
-		   MBEDTLS_PSA_ACCEL_KEY_TYPE_CHACHA20
+	help
+	  This configuration enables the usage of the Oberon PSA driver. Note that the Oberon
+	  PSA core is always included in the build.
```

```suggestion
	  This configuration enables the usage of the Oberon PSA driver. Note that the Oberon
	  PSA core is the only supported core.
```

I hope this is not true. Surely the Oberon PSA Core is only incuded in the build when enabled?

---

I don't think we need the the additional phrase `Note that the Oberon PSA core is always included in the build` in this level. I don't know if we get much value on the suggested note either... I would rather want to see an explanation that this is a optimized SW library that implements crypto algorithms

---

I don't understand why the author added this either.

---

I wanted to just make sure that users understand the difference between the Oberon PSA driver and the Oberon PSA core. It seems that it doesn't help much so I will remove the note altogether.